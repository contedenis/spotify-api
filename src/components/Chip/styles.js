// @packages
import styled from 'styled-components';

// @app
import Text from 'components/Text';

const ChipStyled = styled.div`
  border-radius: 25px;
  background: rgb(255,255,255, 0.8);
  padding: 5px 10px;
  width: fit-content;
`;

const TextStyled = styled(Text)`
  ${({ ellipsis }) => ellipsis && `
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis; 
  `};
`;

export {
  ChipStyled,
  TextStyled,
};
