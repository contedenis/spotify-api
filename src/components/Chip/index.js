// @packages
import React from 'react';
import PropTypes from 'prop-types';

// @app
import Text from 'components/Text';

// @own
import { ChipStyled } from './styles';

function Chip({ className, text, size }) {
  return (
    <ChipStyled className={className}>
      <Text type="h5" color="black" size={size}>{text}</Text>
    </ChipStyled>
  );
}

Chip.defaultProps = {
  className: '',
};

Chip.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
};

export default Chip;
