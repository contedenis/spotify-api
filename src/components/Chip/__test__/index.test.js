// @packages
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// @own
import Chip from '../index';

describe('Chip', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Chip />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('render Chip', () => {
    render(<Chip text="chip" />);

    const chipItem = screen.getByRole('heading', { name: 'chip' });
    expect(chipItem).toBeInTheDocument();
  });
});
