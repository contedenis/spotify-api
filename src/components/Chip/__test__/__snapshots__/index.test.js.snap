// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip matches snapshot 1`] = `
<div
  class="sc-gtssRu kPXVJF"
>
  <h5
    class="sc-bdnylx iOBUJT sc-dlnjPT byzczw"
    color="black"
    size="12"
  />
</div>
`;
